### PR TITLE
Document the other ways NML touches neos

### PR DESCRIPTION
**Pull request by zkxs**
sls asked for this https://discord.com/channels/901126079857692714/901129181453230130/1003774038675574845

### CODE REVIEW

**Review by EIA485 — APPROVED** [review]

File: doc/how_nml_works.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Going Into Detail: How NeosModLoader Interfaces with Neos
 
-NeosModLoader only interfaces with Neos in two places: the hook and the compatibility hash.
+NeosModLoader with Neos in two main places: the hook and the compatibility hash.
```
Comment (XDelta) [inline doc/how_nml_works.md:3]: Seems to be missing 'interfaces with'

File: doc/how_nml_works.md [hunk 2]
```diff
@@ -32,3 +32,8 @@ In order to make a client using NeosModLoader compatible with vanilla Neos the c
 [components]: https://wiki.neos.com/Component
 [logix]: https://wiki.neos.com/LogiX
 [data model]: https://wiki.neos.com/Core_Concepts#Data_Model
+
+## Other Minor Ways NeosModLoader Touches Neos
```
Comment (XDelta) [inline doc/how_nml_works.md:36]: Reads weird, may want to reuse 'Interfaces with' or list it as addition hooks.
Alternatively could refer to the main ones as 'Primary' and these as 'Secondary' hooks. 
Primary as required to function
Secondary as QoL/additional features

**Review by XDelta — CHANGES_REQUESTED** [review]

File: doc/how_nml_works.md (hunk 1 above)
Comment (zkxs) [reply doc/how_nml_works.md]: Fixed in [be2481e](https://github.com/neos-modding-group/NeosModLoader/pull/54/commits/be2481e164be748596ab0b66ea6a63dcbd29bea3)

File: doc/how_nml_works.md (hunk 2 above)
Comment (zkxs) [reply doc/how_nml_works.md]: Fixed in  3c4e40d0

**Review by ljoonal — APPROVED** [review]

**Pull request merged**